### ISSUE TITLE
Setting attribute of typing.NamedTuple is not detected

### ISSUE THREAD

**Issue by ramkumarhn** [pylint-dev/pylint#2858]
### Steps to reproduce
```python
from typing import NamedTuple

class Car(NamedTuple):
    color: str
    mileage: float
    automatic: bool

car1 = Car('red', 3812.3, True)
car1.mileage = 12
```

### Current behavior

Currently, at runtime the below exception is thrown.
`car1.mileage = 12`
`AttributeError: can't set attribute`
### Expected behavior
This error should be flagged 

### pylint --version output
pylint 2.3.1
astroid 2.2.5
Python 3.7.2 (default, Jan 13 2019, 12:50:01) 
[Clang 10.0.0 (clang-1000.11.45.5)]

**Comment by Akshay9715** [comment]
Hi 👋, I’d like to work on this issue. I plan to add a brain tip in Astroid to handle immutability for typing.NamedTuple so Pylint can detect invalid assignments. Please assign this to me. Thanks!

**Comment by Pierre-Sassoulas** [comment]
Sounds good to me, I'm assigning you the issue !
